marimo/_server/api/middleware.py

```diff
@@ -360,9 +360,10 @@ async def __call__(
             if self.path_rewrite:
                 ws_path = self.path_rewrite(ws_path)
             ws_url = urljoin(ws_target_url, ws_path)
-            if scope["scheme"] in ("http", "ws"):
+            scheme = urlparse(ws_url).scheme
+            if scheme in ("http", "ws"):
```

Comment:
If the target scheme is already ws or wss, I don't think it needs to be replaced?

Also, that might lead to an unexpected replacement with a target such as `ws://example.com/http` which would be replaced to `ws://example.com/ws`.

Reply:
you are right. i will fix this